Client/Frontend/Browser/Authenticator.swift

```diff
@@ -39,7 +39,32 @@ class Authenticator {
         // Otherwise, try to look them up and show the prompt.
         if let loginsHelper = loginsHelper {
             return loginsHelper.getLoginsForProtectionSpace(challenge.protectionSpace).bindQueue(dispatch_get_main_queue()) { res in
-                let credentials = res.successValue?[0]?.credentials
+
+                var credentials: NSURLCredential? = nil
+                if let logins = res.successValue {
+
+                    // It is possible that we might have duplicate entries since we match against host and scheme://host.
+                    // This is a side effect of https://bugzilla.mozilla.org/show_bug.cgi?id=1238103.
+                    if logins.count > 1 {
+                        credentials = findValidLoginAmongstLogins(logins, forProtectionSpace: challenge.protectionSpace)?.credentials
+                        removeMalformedLoginsFromLogins(logins, usingHelper: loginsHelper)
```

Comment:
This is async, so:

`.upon { log.debug("Removed malformed logins. Success: \($0.isSuccess)") }`